crates/derive/src/combine_api.rs

```diff
@@ -183,10 +201,27 @@ pub fn drain_combiner(
 
         // Send packed key.
         let begin = arena.len();
+        // Update arena_len with each component of the key so that we can assert that every
+        // component of the key extends the arena by at least one byte.
+        let mut arena_len = begin;
```

Comment:
nit: prev_arena_len ?

I think we should remove the null_count stuff.

Also include a reference to the issue for future code archeologists wondering why the heck we're bothering to do this.

Reply:
done